File: src/main/java/edu/berkeley/bid/bullet/CameraImageData.java
package edu.berkeley.bid.bullet;

import java.io.Serializable;

public class CameraImageData implements Serializable {
    private static final long serialVersionUID = -472381267164873L;


    public int m_pixelWidth;
    public int m_pixelHeight;
    public byte [] m_rgbColorData;
    public float [] m_depthValues;
    public int [] m_segmentationMaskValues;

    public CameraImageData (int width, int height) {
	m_pixelWidth = width;
	m_pixelHeight = height;
	int size = width * height;
	
	m_rgbColorData = new byte[size*3];
	m_depthValues = new float[size];
	m_segmentationMaskValues = new int[size];
    }

    public CameraImageData () {
    }

    @Override
    public String toString () {
	return "CameraImageData[width=" + m_pixelWidth + ",height=" + m_pixelHeight + "]";
    }

}

File: src/main/scala/BIDMach/rl/environments/bullet/CameraImageData.scala
package BIDMach.rl.environments.bullet;
import BIDMat.{BMat,FMat,IMat}

@SerialVersionUID(452462543275423L)
class CameraImageData(jcid:edu.berkeley.bid.bullet.CameraImageData) extends Serializable {
    val javaCameraImageData = jcid;

    def width = javaCameraImageData.m_pixelWidth;
    def height = javaCameraImageData.m_pixelHeight;

    val rgb = new BMat(width, height, javaCameraImageData.m_rgbColorData);
    val depth = new FMat(width, height, javaCameraImageData.m_depthValues);
    val segmentation = new IMat(width, height, javaCameraImageData.m_segmentationMaskValues);

    def unpack() = {
	(rgb, depth, segmentation)
    }

    override def toString () = {
	"CameraImageData[width=" + width + ",height=" + height + "]";
    }

}
